# Data Understanding 
In this notebook, the dataset is explored to understand its structure, contents, and overall quality before performing any analysis.
#### The objectives of this notebook are:
- Load and inspect the dataset
- Understand available features
- Determine dataset dimensions
- Examine data types
- Identify potential data quality issues

In [1]:
# importing libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# loading dataset

df=pd.read_csv("../Data/raw/global_financial_markets_2000_Now.csv")

## Data Overview

In [3]:
df.head()

,date,open,high,low,close,volume,symbol,asset_name,asset_type,region
0,2000-01-03,1469.2500,1478.0000,1438.3600,1455.2200,931800000,^GSPC,S&P500,Stock Index,Global
1,2000-01-03,4186.1899,4192.1899,3989.7100,4131.1499,1510070000,^IXIC,NASDAQ,Stock Index,Global
2,2000-01-03,17057.6992,17426.1602,17057.6992,17369.6309,0,^HSI,HSI,Stock Index,Global
3,2000-01-03,0.0098,0.0099,0.0097,0.0098,0,JPYUSD=X,JPY/USD,Currency,Global
4,2000-01-03,5209.5400,5384.6602,5209.5400,5375.1099,0,^BSESN,SENSEX,Stock Index,Global


In [4]:
# getting more details

df.shape

(187611, 10)

In [5]:
df.columns

Index(['date', 'open', 'high', 'low', 'close', 'volume', 'symbol',
       'asset_name', 'asset_type', 'region'],
      dtype='str')

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 187611 entries, 0 to 187610
Data columns (total 10 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   date        187611 non-null  str    
 1   open        187611 non-null  float64
 2   high        187611 non-null  float64
 3   low         187611 non-null  float64
 4   close       187611 non-null  float64
 5   volume      187611 non-null  int64  
 6   symbol      187611 non-null  str    
 7   asset_name  187611 non-null  str    
 8   asset_type  187611 non-null  str    
 9   region      187611 non-null  str    
dtypes: float64(4), int64(1), str(5)
memory usage: 14.3 MB


### Data Overview Summary
The dataset contains **187,611 rows** and **10 columns** detailing global financial market metrics from **January 2000 to May 2026**. It captures daily price metrics (`open`, `high`, `low`, `close`, `volume`) across multiple global asset classes including stock indices, currencies, commodities, and cryptocurrencies.

## Statistical Overview

In [7]:
df.describe()

,open,high,low,close,volume
count,187611.000000,187611.000000,187611.000000,187611.000000,1.876110e+05
mean,4408.078092,4444.787018,4366.875968,4407.227970,1.247674e+09
std,10646.457796,10750.171471,10531.420024,10644.700449,5.489795e+09
min,-14.000000,0.001200,-40.320000,-37.630000,0.000000e+00
25%,1.152600,1.160200,1.143150,1.152500,0.000000e+00
50%,125.060000,128.666100,122.100000,125.126700,7.182200e+04
75%,3343.026000,3375.043450,3310.429950,3342.126450,2.232844e+08
max,124752.140600,126198.070300,123196.046900,124752.531200,3.509679e+11


### Statistical Overview Summary
The summary statistics provide an initial understanding of the distribution and range of numerical variables present in the dataset.

## Initial Conclusion

The dataset was successfully loaded and explored to understand its structure and contents.

#### Key findings:

- The dataset contains 187,611 observations and 10 features.
- Both numerical and categorical variables are present.
- Historical financial market data is available across multiple asset classes and regions.
- The dataset appears suitable for further preprocessing and analysis.

The next notebook will focus on data cleaning and preprocessing.